### ISSUE TITLE
Not all axioms are exported via Hets server

### ISSUE THREAD

**Issue by mcodescu** [spechub/Hets#1609]
The graph obtained when calling ```curl localhost:8000/dg/http://148.251.85.37:8300/static/hetsfile10935.dol ``` has just 2 axioms for the node Generalisation0, but in the specification there are 5 axioms. This is also visible here http://ontohub.org/sandbox/hetsfile10935//Generalisation0///axioms

**Comment by tillmo** [comment]
@cmaeder, do you have an idea here?

**Comment by cmaeder** [comment]
I  cannot analyse this file

```
maeder@lati:~/Hets$ ./hets -g hetsfile10935.dol 
Analyzing file hetsfile10935.dol as library hetsfile10935
Analyzing spec hetsfile10935?Atom
Analyzing spec hetsfile10935?Solar
Analyzing spec hetsfile10935?Generalisation0
/home/maeder/Hets/hetsfile10935.dol:80.54: *** Error:
unexpected mixfix token: <
/home/maeder/Hets/hetsfile10935.dol:81.18: *** Error:
unexpected mixfix token: >
/home/maeder/Hets/hetsfile10935.dol:82.39: *** Error:
unexpected mixfix token: >
hets: user error (Stopped due to errors)
```

which branch am I supposed to use?

**Comment by cmaeder** [comment]
After using `<` and `>` as prefix operators, I see 5 axioms in Generalisation0 (also in my xml output)

```
. <=(0, 0) %(Ax1)%

forall G_G910393 : Int . <=(0, G_G910393) => <=(0, suc(G_G910393))
                                                           %(Ax2)%

forall G_G908948, G_G908857 : Int
. <(G_G908857, G_G908948) <=> >(G_G908948, G_G908857)      %(Ax3)%

. >(mass(G_G907993), mass(G_G907842)) %(Ax4)%

. >(force(G_G913050, G_G907842, G_G907993), 0) %(Ax5)%
```

**Comment by cmaeder** [comment]
```
curl localhost:8000/dg/file%3A%2f%2f%2fhome%2fmaeder%2fHets%2fhetsfile10935.dol
```

with a running server produces the same output as `./hets -v2 -o xml hetsfile10935.dol`

**Comment by cmaeder** [comment]
I could reproduce your error for your original wrong `hetsfile10935.dol`! The hets server seems to ignore the parsing error silently, but don't ask me why.

**Comment by cmaeder** [comment]
Maybe errors are not "forced" (aka evaluated).

**Comment by cmaeder** [comment]
In `PGIP.Server` I guess that `liftIO $ runResultT $ ... getHetsResult` ignores errors. (There are several occurrences.) From a `Result` also `hasErrors` need to be tested, are maybe `resultToMonad` should be used.

**Comment by cmaeder** [comment]
Alternatively, a wrapper around `anaSourceFile` could make the `Result` "more fatal" (by changing the `maybeResult` to `Nothing`, if `diags` `hasErrors`).

**Comment by cmaeder** [comment]
BTW, most errors should not be "fatal" to allow collecting more errors.

**Comment by cmaeder** [comment]
Obviously now, axioms with errors have been simply ignored.

**Comment by mcodescu** [comment]
This was fixed by a7ea46d4c76b11735234092d0a50efcbc56f9642, which has been merged. Now the errors are properly propagated to the Hets server.
